### ISSUE TITLE
Error when importing -> string | URL cannot be assigned to string

### ISSUE THREAD

**Issue by avalero** [eveningkid/denodb#78]
Just importing on a clean project get this error

``` 
error: TS2345 [ERROR]: Argument of type 'string | URL' is not assignable to parameter of type 'string'.
api_1      |   Type 'URL' is not assignable to type 'string'.
api_1      |   return new URL(url).pathname
api_1      |                  ~~~
api_1      |     at https://deno.land/x/dex/lib/deps/path/win32.ts:911:18
```
on `deno 1.2.0` ran on `debian 10` with `denon`

**Comment by Frenzoid** [comment]
Hello avalero!

I've checked this error since im getting it with other imports as well, aside from denodb, and seems to be happening because the current version of deno (1.2.0) doesn't work properly with some old std / library versions (in my case posix and dex libraries) that the current importing library is importing.

So, until denodb's imports are updated and tested, i'd suggest you to stick with previous deno version for now.

Best regards!

**Comment by avalero** [comment]
Thanks!

**Comment by Frenzoid** [comment]
Hello again!

I've gotten a bit more in depth with it and i think the issue might the other way, since the imports on your error "https://deno.land/x/dex/lib/deps/path/win32.ts" are from the master branch, and not a specific version of the dex library as recomended when importing stds / modules, so if they updated it, it might have changed something that throws errors on denodb when running on deno 1.2.0.

Anyway, it might be something different, so dont take my word for granted! But im suspecting it must be something involved with that since it works properly on the previous deno version.

Best regards!

**Comment by avalero** [comment]
Hello @Frenzoid , FYI, in case of the deno mongo driver, with deno 1.2.0 you need to use deno-mongo 0.9.1 version. Previous version is no longer supported.
I hope this can help.
BR

**Comment by Frenzoid** [comment]
I see, thanks a lot! :).

**Comment by ToWeLong** [comment]
I try to use deno's 1.1.3 version,but it has error also,and I switched back to version 1.2.0
`error: TS2345 [ERROR]: Argument of type 'string | URL' is not assignable to parameter of type 'string'.
  Type 'URL' is not assignable to type 'string'.
  return new URL(url).pathname
                 ~~~
    at https://deno.land/x/dex/lib/deps/path/win32.ts:911:18

TS2345 [ERROR]: Argument of type 'string | URL' is not assignable to parameter of type 'string'.
  Type 'URL' is not assignable to type 'string'.
  return new URL(url).pathname;
                 ~~~
    at https://deno.land/x/dex/lib/deps/path/posix.ts:433:18

Found 2 errors.`

**Comment by ToWeLong** [comment]
Do you have any specific solutions？

**Comment by eveningkid** [comment]
@ToWeLong well for now we don't as we are still waiting on dex to be fixed (https://github.com/denjucks/dex/issues/6).

I might take a look at it if it doesn't get upgraded soon... Or if any of you wants to give it a try (it looks simple enough I guess), please do.

I just don't have much time recently and struggle at least getting back to you guys on this project. I just wish I wouldn't have to fix other people's projects too otherwise that would be too much.

Anyways, we need to wait, or PR on dex

**Comment by eveningkid** [comment]
Hey guys,

I just released 1.0.6 which is also available on `deno.land` now.

On a clean project, it works fine for me so that should do it.

Let me know if it doesn't (but let's be hopeful!).

Have a great week :)

**Comment by angel-codes** [comment]
> Hey guys,
> 
> I just released 1.0.6 which is also available on `deno.land` now: http://deno.land/x/denodb@v1.0.6.
> 
> On a clean project, it works fine for me so that should do it.
> 
> Let me know if it doesn't (but let's be hopeful!).
> 
> Have a great week :)

I have the same error when I import my database configuration in my entry file. What is the solution? I use Deno 1.2.3

>  error: TS2345 [ERROR]: Argument of type 'string | URL' is not assignable to parameter of type 'string'.
>   Type 'URL' is not assignable to type 'string'.
>   return new URL(url).pathname
>                  ~~~
>     at https://deno.land/x/dex@1.0.2/lib/deps/path/win32.ts:911:18
> 
> TS2345 [ERROR]: Argument of type 'string | URL' is not assignable to parameter of type 'string'.
>   Type 'URL' is not assignable to type 'string'.
>   return new URL(url).pathname;
>                  ~~~
>     at https://deno.land/x/dex@1.0.2/lib/deps/path/posix.ts:433:18
> 
> Found 2 errors.

**Comment by eveningkid** [comment]
Hi Angel,

You are right, I actually completely forgot to upgrade my deno version!

After doing so, I upgraded to the latest version of `dex` which now comes from Github, not deno.land anymore.

If you install 1.0.7, it should now work: https://deno.land/x/denodb@v1.0.7.

Hope this helps :)

**Comment by angel-codes** [comment]
Me again, maybe I'm doing something wrong... the error persists for my, I updated denodb to the version that you tell me, and this is my logs:

> Something went wrong... TypeError: TS2345 [ERROR]: Argument of type 'string | URL' is not assignable to parameter of type 'string'.
>   Type 'URL' is not assignable to type 'string'.
>   return new URL(url).pathname
>                  ~~~
>     at https://deno.land/std@0.50.0/path/win32.ts:911:18
> 
> TS2345 [ERROR]: Argument of type 'string | URL' is not assignable to parameter of type 'string'.
>   Type 'URL' is not assignable to type 'string'.
>   return new URL(url).pathname;
>                  ~~~
>     at https://deno.land/std@0.50.0/path/posix.ts:433:18
> 
> Found 2 errors.

You can check my project, maybe I'm the error haha, all of the things still in the database connection when I import the configuration file in the entry file (configuration, table link, and database sync). Here is the link of my [repo](https://github.com/angel-codes/dino-api/tree/develop)

**Comment by eveningkid** [comment]
Hi Angel,

No problems, I think I know what the issue is. In your dependencies file, you should replace:
```json
"denodb": "https://deno.land/x/denodb/mod.ts",
```
With:
```json
"denodb": "https://deno.land/x/denodb@v1.0.7/mod.ts",
```

Otherwise you can keep the master branch as a default (which you have right now), you just need to **refresh your local denodb package** using the `--reload` option:

```bash
deno run --your-options-as-usual --reload index.ts
```

Hope it works for your this time :)

**Comment by avalero** [comment]
You might be having a problem with cache. Run deno with the `--reload` flag

**Comment by angel-codes** [comment]
Well, this happens: When I import my database configuration, and I run the server (with the --reload flag) the server starts downloading some modules, and then, the server starts to run and they show the first log (Oak event) and then, they start to download more modules (denodb modules) when the download finishes they throw the same error (log of the promise in database.ts:23) .


> Server running in http://localhost:5000
> Download https://deno.land/x/denodb@v1.0.7/unstable_deps.ts
> Download https://raw.githubusercontent.com/eveningkid/deno_mongo/master/mod.ts
> Download https://raw.githubusercontent.com/eveningkid/deno_mongo/master/ts/client.ts
> Download https://raw.githubusercontent.com/eveningkid/deno_mongo/master/ts/collection.ts 
> Download https://raw.githubusercontent.com/eveningkid/deno_mongo/master/ts/database.ts   
> Download https://raw.githubusercontent.com/eveningkid/deno_mongo/master/ts/result.ts     
> Download https://raw.githubusercontent.com/eveningkid/deno_mongo/master/ts/types.ts      
> Download https://raw.githubusercontent.com/eveningkid/deno_mongo/master/ts/util.ts
> Download https://raw.githubusercontent.com/eveningkid/deno_mongo/master/ts/type_convert.ts
> Download https://raw.githubusercontent.com/eveningkid/deno_mongo/master/deps.ts
> Download https://deno.land/x/plugin_prepare@v0.6.0/mod.ts
> Download https://deno.land/x/plugin_prepare@v0.6.0/deps.ts
> Download https://deno.land/std@v0.50.0/fs/exists.ts
> Download https://deno.land/std@v0.50.0/log/mod.ts
> Download https://deno.land/std@v0.50.0/path/mod.ts
> Download https://deno.land/std@0.50.0/fs/exists.ts
> Download https://deno.land/std@0.50.0/log/mod.ts
> Download https://deno.land/std@0.50.0/path/mod.ts
> Download https://deno.land/std@0.50.0/log/logger.ts
> Download https://deno.land/std@0.50.0/log/handlers.ts
> Download https://deno.land/std@0.50.0/testing/asserts.ts
> Download https://deno.land/std@0.50.0/log/levels.ts
> Download https://deno.land/std@0.50.0/path/win32.ts
> Download https://deno.land/std@0.50.0/path/posix.ts
> Download https://deno.land/std@0.50.0/path/common.ts
> Download https://deno.land/std@0.50.0/path/separator.ts
> Download https://deno.land/std@0.50.0/path/interface.ts
> Download https://deno.land/std@0.50.0/path/glob.ts
> Download https://deno.land/std@0.50.0/fmt/colors.ts
> Download https://deno.land/std@0.50.0/fs/exists.ts
> Download https://deno.land/std@0.50.0/testing/diff.ts
> Download https://deno.land/std@0.50.0/path/_constants.ts
> Download https://deno.land/std@0.50.0/path/_util.ts
> Download https://deno.land/std@0.50.0/path/_globrex.ts
> Download https://deno.land/std@0.50.0/path/mod.ts
> Check https://deno.land/x/denodb@v1.0.7/unstable_deps.ts
> Something went wrong... TypeError: TS2345 [ERROR]: Argument of type 'string | URL' is not assignable to parameter of type 'string'.
>   Type 'URL' is not assignable to type 'string'.
>   return new URL(url).pathname
>                  ~~~
>     at https://deno.land/std@0.50.0/path/win32.ts:911:18
> 
> TS2345 [ERROR]: Argument of type 'string | URL' is not assignable to parameter of type 'string'.
>   Type 'URL' is not assignable to type 'string'.
>   return new URL(url).pathname;
>                  ~~~
>     at https://deno.land/std@0.50.0/path/posix.ts:433:18
> 
> Found 2 errors.

**Comment by eveningkid** [comment]
Alright, seems like it is related to the mongo adapter that I haven't upgraded for awhile.

Could you open another issue and we'll go from there?

Thanks!
